src/api/include/migraphx/migraphx.hpp

```diff
@@ -310,13 +346,21 @@ struct interface_base : Base
 
     protected:
     template <class F>
-    static migraphx_status try_(F f) // NOLINT
+    static migraphx_status try_(F f, const char** ex_msg = nullptr) // NOLINT
     {
         try
         {
             f();
             return migraphx_status_success;
         }
+        catch(std::exception& ex)
+        {
+            if(ex_msg)
+            {
+                *ex_msg = ex.what();
```

Comment:
This will be a dangling reference upon returning. I think you will need to provide pointers to a char array that you copy into.

Reply:
I am using pointer to char_array. Moving `ex.what()` to `*ex_msg`